src/link.ts

```diff
@@ -265,19 +280,65 @@ export class Link implements esr.AbiProvider {
         }
     }
 
+    /**
+     * Sign and optionally broadcast a EOSIO transaction (matching eosjs transact)
+     *
+     * Example:
+     *
+     * ```ts
+     * let result = await myLink.transact(transaction, options)
+     * ```
+     *
+     * @param transaction The transaction.
+     * @param options The eosjs options to use for the transaction.
+     * @param transport Transport override, for internal use.
+     */
+    public async transact(
+        transaction: any,
+        {
+            broadcast = true,
+            sign = true,
+            blocksBehind,
+            expireSeconds
+        }: EosjsTransactArgs = {},
+        transport?: LinkTransport
+    ): Promise<any> {
+        // Assemble data for internal transact method
+        const tx:any = {
+            broadcast,
+            blocksBehind,
+            expireSeconds,
+        }
+        // Translate the eosjs data into the most optimized ESR payload possible
```

Comment:
Didn't realize that! Good to know 👍

Reply:
Why not include `blocksBehind` and `expireSeconds` in the `TransactOptions`?

These options as it stands are still not compatible with eosjs.

Reply:
Hrm I see what you mean about them not being used in link - I think they should be eventually though.

Reply:
Like what would even `sign: false` mean in link? I think it's fine we are compatible with the structure of the call and ignore stuff we can't act on even if it's passed. We could implement the expire and blocks behind but that would only work for transactions (since the wallet chooses tapos and expiry in ESR for everything but a full tx) and when you submit a full tx you already need to set your expiry and tapos values when you created it so hard to see those being useful

Reply:
`sign: false` and `broadcast: true` could allow a full transaction, with signatures, to be passed to the wallet and the wallet could then broadcast to the blockchain? 

That's what we do in the desktop wallet at least for that situation. It's useful for broadcasting cold-wallet signed transactions.

Actually - that could be a pretty cool use case for mobile. If a user has a cold/offline wallet, and they could just scan a completed/signed transaction on the screen of that device and broadcast it.